File: 백준/Silver/10828. 스택/스택.js
function solution(input) {
    const N = Number(input[0]);
    input.shift();
    const stack = [];
    let result = '';
    
    for(let i = 0; i < N; i++) {
        switch(input[i][0]) {
            case 'push': {
                stack.push(input[i][1]);
                break;
            }
            case 'pop': {
                if(stack.length === 0) {
                    result += '-1\n';
                } else {
                    result += stack.pop() + '\n';
                }
                break;
            }
            case 'size': {
                result += stack.length + '\n';
                break;
            }
            case 'empty': {
                if(stack.length === 0) {
                    result += '1\n';
                } else {
                    result += '0\n';
                }
                break;
            }
            case 'top': {
                if(stack.length === 0) {
                    result += '-1\n';
                } else {
                    result += stack[stack.length - 1] + '\n';
                }
                break;
            }
        }
    }

    console.log(result);
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,    
});

let input = [];
let list = [];

rl.on("line", function (line) {
    input.push(line)
}).on("close", function () {
    input.forEach((val) => {
        list.push(val.split(' ').map((el) => (el)));
    });
    solution(list);
    process.exit();
});

File: 백준/Silver/10828. 스택/스택.py
"""
시간 : 0.5초 -> 1000만개 10000000
입력: 10,000개 O(N^2)은 위험할 수 있음

명령은 총 5가지
- push: 스택에 추가
- pop: 가장 위에 있는 정수를 빼고 출력, 빈 경우 -1 출력
- size: 아이템 개수 출력
- empty: 스택이 비어있으면 1, 아니면 0
- top: 가장 마지막에 있는 수 출력, 빈 경우 -1 출력
"""
import sys

class Stack:
    def __init__(self):
        self.stack = []

    def is_empty(self):
        if len(self.stack) == 0:
            return 1
        else:
            return 0

    def push(self, X):
        self.stack.append(X)

    def pop(self):
        if not self.is_empty():
            print(self.stack.pop())
        else:
            print(-1)

    def size(self):
        print(len(self.stack))

    def top(self):
        if not self.is_empty():
            print(self.stack[len(self.stack) - 1])
        else:
            print(-1)

stack = Stack()
N = int(sys.stdin.readline())

for _ in range(N):
    com = sys.stdin.readline()

    if com.startswith("push"):
        command, X = com.split()
        stack.push(X)
    elif com.startswith("pop"):
        stack.pop()
    elif com.startswith("size"):
        stack.size()
    elif com.startswith("empty"):
        print(stack.is_empty())
    elif com.startswith("top"):
        stack.top()